Preservation of OS images when re-runnning 'epicli apply' (#2329)
epiphany-platform/epiphany: docs/home/howto/CLUSTER.md

        key: aws_key
        secret: aws_secret
      use_public_ips: false

>       os_image: default

Reviewer: ```suggestion
      default_os_image: default
```
?

      subscription_name: Subscribtion_name
      use_service_principal: false
      use_public_ips: false

>       os_image: default

Reviewer: ```suggestion
      default_os_image: default
```
?

    Epicli will read this file and automaticly use it for authentication for resource creation and management.

>     For both `aws`and `azure`here the following cloud attributes overlap:

Reviewer 2: ```suggestion
    For both `aws`and `azure` the following cloud attributes overlap:
```

>     - `use_public_ips`: When `true`, the VMs will also have a direct interface to the internet. While this is easy for setting up a cluster for testing it should not be used in production. A VPN setup should be used which we will document in a different section (TODO).

Reviewer 2: ```suggestion
    - `use_public_ips`: When `true`, the VMs will also have a direct interface to the internet. While this is easy for setting up a cluster for testing, it should not be used in production. A VPN setup should be used which we will document in a different section (TODO).
```

>     - `os_image`: Lets you more easily select Epiphany team validated and tested OS images. When one is selected, it will be applied to **every** `infrastructure/virtual-machine` document in the cluster regardless of user defined ones.

Reviewer: please add information about that "preserve" behaviour to upgrade documentation

Author: Jup, but that is not for this location.

>     - `os_image`: Lets you more easily select Epiphany team validated and tested OS images. When one is selected, it will be applied to **every** `infrastructure/virtual-machine` document in the cluster regardless of user defined ones.

Reviewer: ```suggestion
    - `default_os_image `: Lets you more easily select Epiphany team validated and tested OS images. When one is selected, it will be applied to **every** `infrastructure/virtual-machine` document in the cluster regardless of user defined ones.
```
?

>     - `use_public_ips`: When this is true the VM's will also have a direct inferface to the internet. While this is easy for setting up a cluster for testing it should not be used in production. A VPN setup should be used which we will document in a different section (TODO).
>     - `os_image`: This attribute lets you more easily select Epiphany team validated and tested OS images. When one is selected it will be applied to **every** `infrastructure/virtual-machine` document in the cluster regardless of user defined ones.

Reviewer 2: ```suggestion
    - `use_public_ips`: When `true`, the VMs will also have a direct interface to the internet. While this is easy for setting up a cluster for testing it should not be used in production. A VPN setup should be used which we will document in a different section (TODO).
    - `os_image`: Lets you more easily select Epiphany team validated and tested OS images. When one is selected, it will be applied to **every** `infrastructure/virtual-machine` document in the cluster regardless of user defined ones.
```

                  The following values are accepted:

>                   - `default`: Applies user defined `infrastructure/virtual-machine` documents when generating a new configuration.

Reviewer 2: IMO the `default` word can be misleading in this context since "user defined" looks more like `custom` than `default`. 

So the first idea is:
1) Rename `default` to `custom` or `specified`.
2) Use `ubuntu-18.04-x86_64` as the default value for this setting.

Second idea could be:
1) Rename `os_image` property to `default_os_image` or `global_os_image` or `force_os_image`.
2) If property from point 1 is present, apply its value, otherwise use user defined `infrastructure/virtual-machine` documents.

These are just my personal thoughts, not a request, so feel free to ignore them 😉

Author: I think first idea does not make sense imo, also renaming to ubuntu-18.04-x86_64 will not work because then you will have issues with backwards compatibility. I  think os_image to default_os_image will be good so will make the change.

Author: Done.

>                   - `default`: The default value when generating a new configuration which will result in the use of user defined `infrastructure/virtual-machine` documents.

Reviewer 2: ```suggestion
                  - `default`: Applies user defined `infrastructure/virtual-machine` documents when generating a new configuration.
```

                  - `ubuntu-18.04-x86_64`: Applies the latest validated and tested Ubuntu 18.04 image to all `infrastructure/virtual-machine` documents on `x86_64` on Azure and AWS.
                  - `redhat-7-x86_64`: Applies the latest validated and tested RedHat 7.x image to all `infrastructure/virtual-machine` documents on `x86_64` on Azure and AWS.
                  - `centos-7-x86_64`: Applies the latest validated and tested CentOS 7.x image to all `infrastructure/virtual-machine` documents on `x86_64` on Azure and AWS.

>                   - `centos-7-arm64`: Applies the latest validated and tested CentOS 7.x image to all `infrastructure/virtual-machine` documents on `arm64` on AWS. Azure currently does support `arm64`.

Reviewer 2: ```suggestion
                  - `centos-7-arm64`: Applies the latest validated and tested CentOS 7.x image to all `infrastructure/virtual-machine` documents on `arm64` on AWS. Azure currently doesn't support `arm64`.
```
